app/src/components/progressbar.tsx

```diff
@@ -79,98 +85,149 @@ const Bookmarks = ({
   };
 
   return (
-    <Container fluid className="fixed-bottom py-0">
-      <Row className="justify-content-center position-relative">
-        {Object.entries(bookmarks).map(([number, name]) => {
-          const position = (parseInt(number) / length) * 100;
-          return (
-            <OverlayTrigger
-              key={`bookmark-${number}`}
-              placement="top"
-              delay={{ show: 0, hide: 100 }}
-              overlay={<Tooltip>{name}</Tooltip>}
-            >
-              <Col
-                xs="auto"
-                style={{
-                  position: "absolute",
-                  left: `${position}%`,
-                  bottom: "14px",
-                  transform: "translateX(-50%)",
-                }}
+    <Row className="position-relative">
+      {[...Array(length).keys()].map((position) => {
+        const isDisabled = disabledFrames.includes(position);
+        const commonStyles = {
```

Comment:
test without the dynamic parts and see if still slow. If not maybe it is somewhat fine to use this, altough if we want to view very large trajectories directly from h5 this can and will be 100k+